File: ruby/resistor-color-duo/resistor_color_duo.rb
=begin
Write your code for the 'Resistor Color Duo' exercise in this file. Make the tests in
`resistor_color_duo_test.rb` pass.

To get started with TDD, see the `README.md` file in your
`ruby/resistor-color-duo` directory.
=end

class ResistorColorDuo

  COLOR_BANDS = %w[black brown red orange yellow green blue violet grey white]

  def self.value(arr)
    "#{COLOR_BANDS.index(arr[0])}#{COLOR_BANDS.index(arr[1])}".to_i
  end

end

File: javascript/resistor-color-duo/resistor-color-duo.js
const COLORS = ['black', 'brown', 'red', 'orange', 'yellow', 'green', 'blue', 'violet', 'grey', 'white']

// export const decodedValue = (arr) => +`${COLORS.indexOf(arr[0])}${COLORS.indexOf(arr[1])}`

//Two suggestions from mentor: deconstructed array in argument; using math instead of casting to string and back to number

export const decodedValue = ([color1, color2]) => COLORS.indexOf(color1) * 10 + COLORS.indexOf(color2);